ui/guest/src/controls/rte.ts

```diff
@@ -24,32 +24,72 @@ import { GuestStandardAction } from '../store/models/GuestStandardAction';
 import { Observable, Subject } from 'rxjs';
 import { startWith } from 'rxjs/operators';
 import $ from 'jquery';
+import LookupTable from '@craftercms/studio-ui/models/LookupTable';
+import { reversePluckProps } from '../utils/object';
+// TODO: this needs to be imported from studio-ui
+import { SHOW_EDIT_DIALOG } from '../constants';
 
 export function initTinyMCE(
   record: ElementRecord,
-  validations: Partial<ContentTypeFieldValidations>
+  validations: Partial<ContentTypeFieldValidations>,
+  rteConfig?: LookupTable<any>,
+  site?: string
 ): Observable<GuestStandardAction> {
   const dispatch$ = new Subject<GuestStandardAction>();
   const { field } = iceRegistry.getReferentialEntries(record.iceIds[0]);
   const type = field?.type;
-  const plugins = ['paste'];
+  const plugins = 'paste';
   const elementDisplay = $(record.element).css('display');
   if (elementDisplay === 'inline') {
     $(record.element).css('display', 'inline-block');
   }
+
+  const openEditForm = () => {
+    post({
+      type: SHOW_EDIT_DIALOG,
+      payload: {
+        selectedFields: [field.id]
+      }
+    });
+  };
+
+  const controlPropsMap = {
+    enableSpellCheck: 'browser_spellcheck',
+    forceRootBlockPTag: 'forced_root_block'
+  };
+  const controlProps = {};
+  Object.keys(controlPropsMap).forEach((key) => {
+    if (field.properties?.[key]) {
+      const propKey = controlPropsMap[key];
+      controlProps[propKey] = field.properties[key].value;
+    }
+  });
+
+  const external = {
+    acecode: '/studio/static-assets/js/tinymce-plugins/ace/plugin.min.js',
+    editform: '/studio/static-assets/js/tinymce-plugins/editform/plugin.js'
+  };
+
+  if (rteConfig?.tinymceOptions?.external_plugins) {
+    Object.entries(rteConfig.tinymceOptions.external_plugins).forEach((entry) => {
+      external[entry[0]] = (entry[1] as string).replaceAll('{site}', site);
+    });
+  }
```

Comment:
Expanding the `{site}` expression should be done at the reducer when parsing the config.
```suggestion
  const external = {
    ...rteConfig?.tinymceOptions?.external_plugins,
    acecode: '/studio/static-assets/js/tinymce-plugins/ace/plugin.min.js',
    editform: '/studio/static-assets/js/tinymce-plugins/editform/plugin.js'
  };
```